src/controller/ReadInteraction.h

```diff
@@ -46,37 +46,33 @@ CHIP_ERROR ReportAttribute(Messaging::ExchangeManager * exchangeMgr, EndpointId
                            AttributeId attributeId, ReportAttributeParams<DecodableAttributeType> && readParams)
 {
     app::AttributePathParams attributePath(endpointId, clusterId, attributeId);
-    app::ReadClient * readClient         = nullptr;
     app::InteractionModelEngine * engine = app::InteractionModelEngine::GetInstance();
     CHIP_ERROR err                       = CHIP_NO_ERROR;
 
     readParams.mpAttributePathParamsList    = &attributePath;
     readParams.mAttributePathParamsListSize = 1;
 
     auto onDone = [](app::ReadClient * apReadClient, TypedReadAttributeCallback<DecodableAttributeType> * callback) {
+        chip::Platform::Delete(apReadClient);
         chip::Platform::Delete(callback);
     };
 
     auto callback = chip::Platform::MakeUnique<TypedReadAttributeCallback<DecodableAttributeType>>(
         clusterId, attributeId, readParams.mOnReportCb, readParams.mOnErrorCb, onDone, readParams.mOnSubscriptionEstablishedCb);
     VerifyOrReturnError(callback != nullptr, CHIP_ERROR_NO_MEMORY);
 
-    ReturnErrorOnFailure(engine->NewReadClient(&readClient, readParams.mReportType, &(callback->GetBufferedCallback())));
+    auto readClient = chip::Platform::MakeUnique<app::ReadClient>(engine, exchangeMgr, *callback.get(), readParams.mReportType);
```

Comment:
This lost the GetBufferedCallback. Luckily, we have tests, and they are catching this...